src/test/java/org/opensearch/security/filter/SecurityRestFilterTests.java

```diff
@@ -300,4 +303,67 @@ public void testAllowlistedApiWithoutExtraSlash() throws Exception {
         );
         assertThat(response.getBody(), response.getStatusCode(), equalTo(HttpStatus.SC_FORBIDDEN));
     }
+
+    /**
+     * Tests that when has_permission_check param is absent;
+     * the normal request flow is executed.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testWithoutHasPermissionCheckParam() throws Exception {
+        setup();
+
+        // ALLOWLIST GET /_cluster/health
+        rh.keystore = "restapi/kirk-keystore.jks";
+        rh.sendAdminCertificate = true;
+        response = rh.executePutRequest(
+            "_plugins/_security/api/allowlist",
+            "{\"enabled\": true, \"requests\": {\"/_cluster/health\": [\"GET\"]}}",
+            nonAdminCredsHeader
+        );
+
+        rh.sendAdminCertificate = false;
+        assertThat(response.getStatusCode(), equalTo(HttpStatus.SC_OK));
+        assertFalse(response.getBody().contains("\"accessAllowed\""));
+        assertFalse(response.getBody().contains("\"missingPrivileges\""));
+
+    }
+
+    /**
+     * Tests that the has_permission_check param works correctly.
+     * When has_permission_check=true is added to a request, returns
+     * whether the request would be allowed or Denied, without actually executing the request.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testHasPermissionCheckParam() throws Exception {
+        setup();
+
+        rh.keystore = "restapi/kirk-keystore.jks";
+        rh.sendAdminCertificate = true;
+        response = rh.executeGetRequest(
+            "_cluster/health?has_permission_check=true",
+            nonAdminCredsHeader
+        );
+        rh.sendAdminCertificate = false;
+        assertThat(response.getBody(), response.getStatusCode(), equalTo(HttpStatus.SC_OK));
+        assertTrue(response.getBody().contains("\"accessAllowed\":"));
```

Comment:
can you check the actual value as well to see if contains true or false, try a different endpoint where it is explicitly false and contains missingPrivileges.